src/utils/mempool.rs

```diff
@@ -138,3 +145,54 @@ impl io::Seek for MemPool {
         io::Seek::seek(&mut self.file, pos)
     }
 }
+
+fn create_shm_fd() -> io::Result<RawFd> {
+    loop {
+        match memfd::memfd_create(
+            CStr::from_bytes_with_nul(b"smithay-client-toolkit\0").unwrap(),
+            memfd::MemFdCreateFlag::MFD_CLOEXEC,
+        ) {
+            Ok(fd) => return Ok(fd),
+            Err(nix::Error::Sys(Errno::EINTR)) => continue,
+            Err(nix::Error::Sys(errno)) => return Err(io::Error::from(errno)),
```

Comment:
Actually, I think we should break on `Err(nix::Error::Sys(Errno::ENOSYS))` (which means that `memfd_create` is not supported by the current kernel, and thus we should fallback on shm_open, if I believe the [GTK code](https://gitlab.gnome.org/GNOME/gtk/blob/master/gdk/wayland/gdkdisplay-wayland.c#L1212-1216)), and the final `Err(_)` catch-all should likely be marked `unreachable!()`, as no nix-specific errors can be generated by this function.

Reply:
Nix::Error can return errors other then the Sys enum which means Err(_) isn't really unreachable

Reply:
I've used panic!(err) instead of unreachable!(err) because its not unreachable and it does the same thing as panic.